docker.go

```diff
@@ -794,7 +794,8 @@ func (p *DockerProvider) BuildImage(ctx context.Context, img ImageBuildInfo) (st
 	err = backoff.Retry(func() error {
 		resp, err = p.client.ImageBuild(ctx, buildContext, buildOptions)
 		if err != nil {
-			if _, ok := err.(errdefs.ErrNotFound); ok {
+			var enf errdefs.ErrNotFound
+			if ok := errors.As(err, &enf); ok {
```

Comment:
One question regarding style: should we stand with this ok pattern, or just use this one?
```suggestion
			if errors.As(err, &enf) {
```

The same would apply to all the changes below.